packages/create-redwood-app/template/web/src/index.js

```diff
@@ -1,18 +1,13 @@
 import ReactDOM from 'react-dom'
 
-import { FatalErrorBoundary } from '@redwoodjs/web'
-import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'
-
-import FatalErrorPage from 'src/pages/FatalErrorPage'
-import Routes from 'src/Routes'
+import App from './App'
 
 import './index.css'
 
-ReactDOM.render(
-  <FatalErrorBoundary page={FatalErrorPage}>
-    <RedwoodApolloProvider>
-      <Routes />
-    </RedwoodApolloProvider>
-  </FatalErrorBoundary>,
-  document.getElementById('redwood-app')
-)
+const rootElement = document.getElementById('redwood-app')
+
+if (rootElement.hasChildNodes()) {
+  ReactDOM.hydrate(<App />, rootElement)
+} else {
+  ReactDOM.render(<App />, rootElement)
+}
```

Comment:
Hmmm... It feels a bit weird that this pre-render logic has to be in this file, but maybe we can restore `index.js` to what it was before and try something like this.

```js
const ReactDOMOperation = process.env.PRERENDERING ? ReactDOM.hydrate : ReactDOM.render
```

Reply:
This isn't really prerendering logic to be honest, it's the standard way of doing hydration. It's not checking whether its prerendering, but checking that if the page already has rendered nodes, to not replace the entire thing but instead attach event handlers / update any outdated DOM elements.

Reply:
Ok, but do we have to split this out into App.js?

It seems like this conditional check is the only reason for doing so and I really liked the symmetry of `index.{html, css, js, ts}` as the entry points for everything.

Reply:
Sure, the main reason for splitting is actually this line here https://github.com/redwoodjs/redwood/pull/1641/files#diff-4127298228691875474bb71bfb91612ef45b7649fdb976e58bec5f5617c2d3aeR54

Because we don't want to run any of the other code there, such as ReactDOM.render, finding the root element, etc.

Totally open to a different way of doing this, but for me App.js didn't seem like the worst option - for some reason I'm thinking its quite a common pattern.

Reply:
@peterp any further thoughts on this? Worth keeping in mind index.js is still there, and it is still the entry point. Only difference is, App.js now gets imported in index.js with the JSX markup